src/asymmetric_cipher.c

```diff
@@ -296,12 +296,28 @@ static int p11prov_rsaenc_decrypt(void *ctx, unsigned char *out, size_t *outlen,
         goto endsess;
     }
 
+    /* Special handling against PKCS#1 1.5 side channel leaking */
+    if (mechanism.mechanism == CKM_RSA_PKCS) {
+        ret = side_channel_free_Decrypt(encctx->provctx, sess, (void *)in,
+                                        inlen, out, &out_size);
+        /* the error case need to be handled in a side-channel free way, so
+         * conditionals ineed to be idempotent. Setting outlen unconditionally
+         * is fine because out_size is initialized to the value of outlen
+         * itself, and the value should not matter in an error condition */
+        *outlen = out_size;
+        if (ret == CKR_OK) {
```

Comment:
this will leak, the compare, the subsequent jump, and the secret data dependent load of constant from memory

Reply:
I added constant time functions for this.